src/app/teams/teams.component.ts

```diff
@@ -0,0 +1,64 @@
+import {Component, OnInit} from '@angular/core';
+import {DataService} from '../data.service';
+import {ActivatedRoute, Router} from '@angular/router';
+import {Team} from '../model/Team';
+
+@Component({
+  selector: 'app-teams',
+  templateUrl: './teams.component.html',
+  styleUrls: ['./teams.component.css']
+})
+export class TeamsComponent implements OnInit {
+
+  constructor(private dataService: DataService,
+              private router: Router,
+              private route: ActivatedRoute) {
+  }
+
+  teamsList = new Array<Team>();
+  selectedTeam: Team = new Team();
+  selectedTeamCrest = 'none';
+
+  message = '';
+  backgroundImageURL: string;
+
+  activeCompetitionNo: number;
+
+
+  ngOnInit(): void {
+    this.processUrlParams();
+  }
+
+  processUrlParams(): void {
+    this.route.queryParams.subscribe(
+      (params) => {
+        this.activeCompetitionNo = params['league_id'];
+        this.backgroundImageURL = 'url(../assets/img/' + this.activeCompetitionNo + '_motive.jpg';
```

Comment:
To zła metoda - zamiast odczytywać URL z paska przeglądarki - powinieneś w routrze określić wrzucić do tego komponentu dane z rodzica który go wywołuje (adnotacje @Input)